rosidl_typesupport_opendds_cpp/resource/srv__type_support.cpp.em

```diff
@@ -245,15 +247,93 @@ void * create_replier__@(service.namespaced_type.name)(
     DDS::Subscriber_var dds_subscriber,
     allocator_t allocator)
 {
-@# TODO: Implement, considering original code in ffe10f9 or earlier
+  @# TODO: remove next line when ready
+  return nullptr;
+
+  using ReplierType = rosidl_typesupport_opendds_cpp::Replier<
+    @(__dds_request_wrapper_msg_type),
+    @(__dds_response_wrapper_msg_type)
+  >;
+
+  @# Register TypeSupports
+  @(__dds_msg_typesupport_type)RequestWrapperTypeSupport_var tsRequest =
+    new @(__dds_msg_typesupport_type)RequestWrapperTypeSupportImpl;
+
+  if (tsRequest->register_type(dds_participant, "@(__dds_request_wrapper_msg_type)_") != DDS::RETCODE_OK) {
+    RMW_SET_ERROR_MSG("request register_type for replier failed");
+    return NULL;
+  }
+
+  @(__dds_msg_typesupport_type)ResponseWrapperTypeSupport_var tsResponse =
+    new @(__dds_msg_typesupport_type)ResponseWrapperTypeSupportImpl;
+
+  if (tsResponse->register_type(dds_participant, "@(__dds_response_wrapper_msg_type)_") != DDS::RETCODE_OK) {
+    RMW_SET_ERROR_MSG("request register_type for replier failed");
+    return NULL;
+  }
+
+  @# Create Topics
+  CORBA::String_var type_name = tsRequest->get_type_name();
+  DDS::Topic_var request_topic = dds_participant->create_topic(request_topic_str,
+                                                      type_name,
+                                                      TOPIC_QOS_DEFAULT,
+                                                      nullptr,
+                                                      OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+  if (!request_topic) {
+    RMW_SET_ERROR_MSG("Request create_topic failed");
     return nullptr;
+  }
+
+  type_name = tsResponse->get_type_name();
+  DDS::Topic_var response_topic = dds_participant->create_topic(response_topic_str,
+                                        type_name,
+                                        TOPIC_QOS_DEFAULT,
+                                        nullptr,
+                                        OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+  if (!response_topic) {
+    RMW_SET_ERROR_MSG("Response create_topic failed");
+    return nullptr;
+  }
+
+  auto _allocator = allocator ? allocator : &malloc;
+  rosidl_typesupport_opendds_cpp::ReplierParams replier_params(dds_participant);
+
+  replier_params.publisher(dds_publisher);
+  replier_params.subscriber(dds_subscriber);
+  replier_params.request_topic_name(request_topic_str);
+  replier_params.reply_topic_name(response_topic_str);
+  replier_params.request_topic(request_topic);
+  replier_params.reply_topic(response_topic);
+
+  ReplierType * replier = static_cast<ReplierType *>(_allocator(sizeof(ReplierType)));
+  try {
+    new (replier) ReplierType(replier_params);
```

Comment:
static_cast?

Reply:
@adamsj-oci this is a placement-new, not a cast